src/osdag/design_type/compression_member/compression.py

```diff
@@ -1566,7 +1571,22 @@ def common_checks_1(self, section, step = 1, list_result = [], list_1 = []):
 
 
     def common_result(self, list_result,result_type):
-            self.result_designation = list_result[result_type]['Designation']
+
+            self.result_designation = list_result[result_type]["Designation"] # TODO debug
+            limit = IS800_2007.cl_3_8_max_slenderness_ratio(1)
+            if self.sec_prop_initial_dict[self.result_designation][2] > limit:
+                logger.warning("Length provided is beyond the limit allowed. [Reference: Cl 3.8, IS 800:2007]")
+                logger.error("Cannot compute. Given Length does not pass for this section.")
+                # self.sec_list.remove(self.section_property.designation )
+            else:
+                logger.info("Length provided is within the limit allowed. [Reference: Cl 3.8, IS 800:2007]" )
+            logger.info(
+                            "The section is {}. The b/t of the section ({}) is {} and d/t is {} and (b+d)/t is {}.  [Reference: Cl 3.7, IS 800:2007].".format(
+                               self.input_section_classification[self.result_designation], self.result_designation,
+                                round(self.width_thickness_ratio, 2), round_up(self.depth_thickness_ratio),
+                                round(self.width_depth_thickness_ratio, 2)))
```

Comment:
Why is there such a deep indentation?

Reply:
As these were pasted from flexure file, so I have left it as it is.